system-tests/e2e-suite/src/test/scala/com/hotels/styx/client/ExpiringConnectionSpec.scala

```diff
@@ -69,25 +81,29 @@ class ExpiringConnectionSpec extends FunSpec
   }
 
   it("Should expire connection after 1 second") {
-    val request = get(styxServer.routerURL("/app1"))
-      .build()
+    val request = get(styxServer.routerURL("/app1")).build()
 
-    val response1 = waitForResponse(client.sendRequest(request))
+    val response1 = waitForSingleResponse(pooledClient.sendRequest(request))
```

Comment:
As previously mentioned, please fix the `waitForResponse` instead of implementing another one.